File: frontend/src/types/followup.ts
export interface FollowUpRecord {
  id: string;
  citizen_id: string;
  assessment_id?: string;
  follow_up_date: string;
  responsible_person: string;
  follow_up_type: 'นัดตรวจเลือดซ้ำ 2-4 สัปดาห์' | 'ติดตามการสวมใส่อุปกรณ์ PPE' | 'ส่งต่อพบแพทย์ รพ.' | 'ติดตามอาการพิษเฉียบพลัน' | 'อื่นๆ';
  result: 'ปกติ/ปลอดภัยแล้ว' | 'ยังมีความเสี่ยง/รอผลตรวจ' | 'ส่งต่อเรียบร้อยแล้ว' | 'ไม่สามารถติดต่อได้';
  notes: string;
  created_at: string;
}

File: backend/models/followup.go
package models

// FollowUpRecord represents a patient/farmer follow-up log entry
type FollowUpRecord struct {
	ID                string `json:"id"`
	CitizenID         string `json:"citizen_id" binding:"required"`
	AssessmentID      string `json:"assessment_id,omitempty"`
	FollowUpDate      string `json:"follow_up_date" binding:"required"`
	ResponsiblePerson string `json:"responsible_person" binding:"required"`
	FollowUpType      string `json:"follow_up_type" binding:"required"`
	Result            string `json:"result" binding:"required"`
	Notes             string `json:"notes"`
	CreatedAt         string `json:"created_at,omitempty"`
}
